BismarkUnchained/scenes/Player/Player.gd

```diff
@@ -35,18 +37,22 @@ func move_translate(delta):
 	
 	# Finding difference between current and target velocity, and adjusting for acceleration
 	var velocity = Vector2(x, y).normalized() * max_velocity
-	if Input.is_action_pressed("player_slowdown"):
-		velocity = velocity.normalized() * slow_velocity
 	var diff_velocity = velocity - curr_velocity
 	var new_velocity = curr_velocity + diff_velocity.normalized() * accel * delta
 	
+	if Input.is_action_pressed("player_slowdown"):
+		velocity = velocity.normalized() * slow_velocity
+		new_velocity = velocity
+	if curr_velocity.length() == max_velocity && abs(velocity.angle() - curr_velocity.angle()) < PI / 2:
+		new_velocity = velocity
+	
 	# Checking that new velocity is not greater magnitude than target velocity, and prevents stuttering at extremely low speeds
 	if new_velocity.length() > velocity.length():
 		new_velocity = velocity
 	
 	# Executing the movement, setting current velocity for next pass, setting global time_delta
 	velocity = new_velocity
-	move_and_collide(velocity * delta)
+	move_and_slide(velocity)
```

Comment:
why not `velocity * delta`?

Reply:
Move_and_slide automatically takes into account delta